airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoDbSourceConfig.java

```diff
@@ -27,33 +29,32 @@
  */
 public record MongoDbSourceConfig(JsonNode rawConfig) {
 
-  /**
-   * Constructs a new {@link MongoDbSourceConfig} from the provided raw configuration.
-   *
-   * @param rawConfig The underlying JSON configuration provided by the connector framework.
-   * @throws IllegalArgumentException if the raw configuration does not contain the
-   *         {@link MongoConstants#DATABASE_CONFIG_CONFIGURATION_KEY} key.
-   */
-  public MongoDbSourceConfig(final JsonNode rawConfig) {
-    if (rawConfig.has(DATABASE_CONFIG_CONFIGURATION_KEY)) {
-      this.rawConfig = rawConfig.get(DATABASE_CONFIG_CONFIGURATION_KEY);
-    } else {
+  public MongoDbSourceConfig {
+    if (rawConfig == null) {
+      throw new IllegalArgumentException("MongoDbSourceConfig cannot accept a null config.");
+    }
+    if (!rawConfig.hasNonNull(DATABASE_CONFIG_CONFIGURATION_KEY)) {
       throw new IllegalArgumentException("Database configuration is missing required '" + DATABASE_CONFIG_CONFIGURATION_KEY + "' property.");
     }
   }
 
+  public JsonNode getDatabaseConfig() {
+    return rawConfig.get(DATABASE_CONFIG_CONFIGURATION_KEY);
```

Comment:
this seems to be the same as what it was before : 
`this.rawConfig = rawConfig.get(DATABASE_CONFIG_CONFIGURATION_KEY);`